src/registry.rs

```diff
@@ -263,6 +263,67 @@ impl Registry {
         Ok(reg)
     }
 
+    /// `prefix` returns a copy of the prefix for the registry, if any.
+    pub fn prefix(&self) -> Option<String> {
+        self.r.read().prefix.clone()
+    }
+
+    /// `set_prefix` sets the prefix for the registry, and returns the previous
+    /// prefix, if any. If `prefix` is empty, the function returns an error.
+    pub fn set_prefix(&self, prefix: &str) -> Result<Option<String>> {
+        if prefix.is_empty() {
+            return Err(Error::Msg("empty prefix namespace".to_string()));
+        }
+
+        let mut core = self.r.write();
+        let result = core.prefix.take();
+        core.prefix = Some(prefix.to_string());
+        Ok(result)
+    }
+
+    /// `clear_prefix` removes the prefix from the registry and returns it to
+    /// the caller, if a prefix was set.
+    pub fn clear_prefix(&self) -> Option<String> {
+        let mut core = self.r.write();
+        core.prefix.take()
+    }
+
+    /// `labels` returns a copy of the common labels for the registry, if any.
+    pub fn labels(&self) -> Option<HashMap<String, String>> {
```

Comment:
Here and in `prefix()` above, can we return a reference instead (and let the consumers clone if they want)?